src/main/java/hashtable/HashTable.java

```diff
@@ -2,22 +2,29 @@
 
 import java.util.ArrayList;
 
+/**
+ * This class implements a hash table, which maps keys to values.
+ * Collision is handled via "Separate Chaining" technique where the main idea is to make each cell of hash table point to a linked list of records that have same hash function value.
```

Comment:
Haven't heard of Separate Chaining. Maybe External Chaining?
Also - we limit lines at 120 symbols.